service/history/shard/context.go

```diff
@@ -117,5 +119,13 @@ type (
 		DeleteWorkflowExecution(workflowKey definition.WorkflowKey, branchToken []byte, version int64) error
 		AddTasks(request *persistence.AddTasksRequest) error
 		AppendHistoryEvents(request *persistence.AppendHistoryNodesRequest, namespaceID namespace.ID, execution commonpb.WorkflowExecution) (int, error)
+
+		GetRemoteAdminClient(cluster string) adminservice.AdminServiceClient
+		GetHistoryClient() historyservice.HistoryServiceClient
+		GetPayloadSerializer() serialization.Serializer
+
+		GetSearchAttributesProvider() searchattribute.Provider
+		GetSearchAttributesMapper() searchattribute.Mapper
+		GetArchivalMetadata() archiver.ArchivalMetadata
```

Comment:
I believe most of them are used in methods like historyEngine.NewEngineWithShardContext and were initially plumbed like shard.GetService()... (that returned resource.Resource) So technically shard still knew about these objects. We can create another container for them, but I'm not clear it is worth it within this PR. Ideally, it's worth reviewing how we use those and refactor the code so that shard doesn't need this ownership at all.